lib/_http_agent.js

```diff
@@ -336,18 +336,18 @@ Agent.prototype.reuseSocket = function reuseSocket(socket, req) {
 };
 
 Agent.prototype.destroy = function destroy() {
-  const sets = [this.freeSockets, this.sockets];
-  for (var s = 0; s < sets.length; s++) {
-    var set = sets[s];
-    var keys = Object.keys(set);
-    for (var v = 0; v < keys.length; v++) {
-      var setName = set[keys[v]];
-      for (var n = 0; n < setName.length; n++) {
-        setName[n].destroy();
-      }
+  destroySockets(this.freeSockets);
+  destroySockets(this.sockets);
+};
+
+function destroySockets(sockets) {
+  for (const setKey in sockets) {
```

Comment:
briefer:
```suggestion
  for (const socket of Object.values(sockets).flat()) {
```

Reply:
But probably also much less performant.

Reply:
`in` changes the behavior though. Before, the prototype would not be checked but now it does.

I do not know why `flat` was suggested. It should already be a flat array?

I think in this specific case it should be fine to use `Object.values`, since it should not be a hot path to destroy the sockets.